src/main/java/core/basesyntax/service/ShopCalculationService.java

```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.FruitShopStorage;
+import core.basesyntax.model.AbstractTransaction;
+import java.util.List;
+
+public interface ShopCalculationService<T> {
+    FruitShopStorage calculate(List<AbstractTransaction<T>> transactions);
```

Comment:
The `calculate` method currently returns a `FruitShopStorage`, which is specific to `Fruit`. To maintain consistency with the generic type `T`, consider changing the return type to `ShopStorage<T>`. This will allow the interface to be more flexible and handle different types of storage.